### DISCUSSION TITLE
How to change the speed and temperature when generating TTS using xtts_v2?

### DISCUSSION THREAD

**Discussion by Zibri** [coqui-ai/TTS#3365] [General Q&A]
```
tts.tts_to_file(text=t, speed=0.8, temperature=0.5, speaker_wav=["references.wav"], langua
ge="en", file_path="test.wav")
```

this does not seem to work.

**Comment by platform-kit** [comment]
@erogol does temp and speed work on XTTS when used locally (not over API)?

**Comment by C00reNUT** [comment]
See the docs https://tts.readthedocs.io/en/dev/models/xtts.html#inference-parameters and adjust the code as stated by the comment # Add custom parameters here https://tts.readthedocs.io/en/dev/models/xtts.html#inference-parameters

**Reply by Zibri** [reply]
I just want to change the speed and temperature.
Is there a way to modify the example below making the result slower and more/less "emotional"?

```
from TTS.api import TTS
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2").to("cpu")
t=open('text.txt', 'r').read().replace('\n','')
tts.tts_to_file(text=t, speaker_wav="./reference.mp3", language="en", file_path="test.wav")
```

**Reply by C00reNUT** [reply]
```
import os
import torch
import torchaudio
from TTS.tts.configs.xtts_config import XttsConfig
from TTS.tts.models.xtts import Xtts

print("Loading model...")
config = XttsConfig()
config.load_json("/path/to/xtts/config.json")
model = Xtts.init_from_config(config)
model.load_checkpoint(config, checkpoint_dir="/path/to/xtts/", use_deepspeed=True)
model.cuda()

print("Computing speaker latents...")
gpt_cond_latent, speaker_embedding = model.get_conditioning_latents(audio_path=["reference.wav"])

print("Inference...")
out = model.inference(
    "It took me quite a long time to develop a voice and now that I have it I am not going to be silent.",
    "en",
    gpt_cond_latent,
    speaker_embedding,
    temperature=0.7, 
    speed=1.0# Add custom parameters here
)
torchaudio.save("xtts.wav", torch.tensor(out["wav"]).unsqueeze(0), 24000)
```
you can change speed and temperature like this, "emotional" characteristics of the model depend partially on the temperature, but more importantly on conditioning_latents aka what kind of wav file audio you use in this part `gpt_cond_latent, speaker_embedding = model.get_conditioning_latents(audio_path=["reference.wav"])`

**Reply by platform-kit** [reply]
@C00reNUT if I'm understanding correctly, the speaker_embedding conditions the voice, while the gpd_cond_latent sets the tone/emotionality -- so would this mean it's possible to generate gpt_cond_latent from a separate piece of audio than that of the speaker, in order to control emotion?

Anyway, back to the question of speed -- adjusting the speed parameter seems to do nothing, in my experience. One would expect that if the default value is 1.0 that a value of 0.5 would be twice as slow. That's not the case.

**Comment by MoctarHaiz** [comment]
Check this Pull Request: [Handle Speed](https://github.com/coqui-ai/TTS/pull/3672)
